packages/common/src/models/Analytics.ts

```diff
@@ -1675,6 +1691,28 @@ type BuyUSDCAddFundsManually = {
   eventName: Name.BUY_USDC_ADD_FUNDS_MANUALLY
 }
 
+// Buy Crypto
+
+type BuyCryptoEvent = {
+  eventName:
+    | Name.BUY_CRYPTO_STARTED
+    | Name.BUY_CRYPTO_ON_RAMP_OPENED
+    | Name.BUY_CRYPTO_ON_RAMP_SUCCESS
+    | Name.BUY_CRYPTO_ON_RAMP_FAILURE
+    | Name.BUY_CRYPTO_ON_RAMP_CANCELED
+    | Name.BUY_CRYPTO_ON_RAMP_CONFIRMED
+    | Name.BUY_CRYPTO_SUCCESS
+    | Name.BUY_CRYPTO_FAILURE
+    | Name.BUY_CRYPTO_RECOVERY_STARTED
+    | Name.BUY_CRYPTO_RECOVERY_FAILURE
+    | Name.BUY_CRYPTO_RECOVERY_SUCCESS
+
+  provider: string
+  amount: number
+  mint: MintName
+  error?: string
```

Comment:
I don't suppose you checked in Amplitude that all these values get sent correctly for each event and that they don't change units or types between events?

Reply:
Yep, they're always from the action's payload, never change!